### DISCUSSION TITLE
Cypress Task issues with Promises

### DISCUSSION THREAD

**Discussion by bradyclifford** [cypress-io/cypress#15313] [General]
I am trying to seed a CosmosDB prior to running a Cypress spec. I am attempting to do so using a "task". When I run the code cosmos code just from node, it works great. But when I import the code, reference it within plugins as a task and run it from a Cypress spec, I get a `Converting circular structure to JSON`. Has something to do with how I am managing the promises.

```
// CosmosDb script
...
export async function createItem(item) {
  const container = getContainer();
  return await container.items.create(item);
}
----------
// Within plugins, creating a Cypress task
const cosmos = require('./cosmos');
...
  async 'db:seed'() {
      const item = { id: '1234', otherData: 'abcd' };
      return cosmos.createItem(item); <-- return a promise

      // return null; <-- if I return a null instead of a promise as above, I don't get the circular reference error
  }
```

**Comment by herecydev** [comment]
I think your bumping into this https://docs.cypress.io/api/commands/task.html#Argument-should-be-serializable

Hope that helps.
